File: solutions/__134_max_profit2.go
package main

// 我们只需把所有上升的差值累加起来即可
func maxProfit(prices []int) int {
	in, prev := 0, 0
	prices = append(prices, -1) // 避免prices是递增 然后没法触发line 13的代码
	n := len(prices)

	profit := 0
	for i := 1; i < n; i++ {
		if prices[prev] > prices[i] {
			profit += prices[prev] - prices[in]
			in = i
		}
		prev = i
	}
	return profit
}

File: solutions/__134_max_profit2.cpp
/*
 *      Author: peanutzhen
 *      Created time: 2021/8/22 13:54:07
 */

#include <bits/stdc++.h>

using namespace std;

// 把 prices 中的所有递增区间累加即可
int maxProfit(vector<int>& prices) {
    int ans = 0;
    int n = (int) prices.size();
    if (n < 2)
        return ans;
    for (int i = 1; i < n; ++i) {
        if (prices[i] > prices[i-1])
            ans += prices[i] - prices[i-1];
    }
    return ans;
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);


    return 0;
}

